### PR TITLE
Bug fixing

### PR DESCRIPTION
**Pull request by stevenpitts**
### Summary

Creates a new test in dev_scripts/HyperdomeConnection.py (https://github.com/arades79/hyperdome/issues/15), fixes a lot of various bugs (the ones necessary to make the test work), specifies or eliminates BaseException uses (https://github.com/arades79/hyperdome/issues/33), eliminates vestige code (https://github.com/arades79/hyperdome/issues/12 and https://github.com/arades79/hyperdome/issues/1).

### Checklist

- [x] If code changes were made then they have been tested.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: dev_scripts/hyperdome_connection.py [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+    def __init__(self):
+        self.onion = None
+        self.app = None
+        self.web = None
+        self.exit_flag = False
+        self.url = None
+        self.t = None
+        self.onion_connected = False
+
+    def prepare_connection(self, num_tries=10):
```
Comment (arades79) [inline dev_scripts/hyperdome_connection.py:53]: We may consider adding tenacity for more robust and generic 'retrying' capabilities in various components

File: dev_scripts/hyperdome_connection.py [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+            return False
+        # Start OnionShare http service in new thread
+        self.t = threading.Thread(target=self.web.start,
+                                  args=(self.app.port, True,
+                                        common.settings.get('public_mode'),
+                                        common.settings.get('slug')),
+                                  daemon=True)
+        self.t.start()
+
+        # TODO this looks dangerously like a race condition
```
Comment (arades79) [inline dev_scripts/hyperdome_connection.py:98]: we should restrict to onion service V3 and remove slugs. This also allows the removal of 'public mode' toggle. Onionshare suggests that slugs are unnecessary for protecting privacy for v3 onions anyway.

File: hyperdome_server/__init__.py [hunk 3]
```diff
@@ -143,6 +143,17 @@ def main(cwd=None):
                 common.settings.save()
 
         print('')
+        url = 'http://{0:s}/{1:s}'.format(app.onion_host, web.slug)
```
Comment (arades79) [inline hyperdome_server/__init__.py:146]: pour salt on slug

File: hyperdome_client/hyperdome_client.py [hunk 4]
```diff
@@ -208,6 +204,13 @@ def send_message(self):
         Send the contents of the message box to the server to be forwarded to
         either counsel or guest.
         """
```
Comment (arades79) [inline hyperdome_client/hyperdome_client.py:206]: this function can be flattened

File: hyperdome_client/hyperdome_client.py [hunk 5]
```diff
@@ -208,6 +204,13 @@ def send_message(self):
...
+        if self.uid and not self.therapist:
+            self.therapist = self.session.post(
+                f"{self.server.url}/request_therapist",
```
Comment (arades79) [inline hyperdome_client/hyperdome_client.py:209]: this seems like it should happen somewhere else? User should be connected to someone before sending a message.

File: hyperdome_server/onion.py [hunk 6]
```diff
@@ -580,13 +578,13 @@ def start_onion_service(self, port):
                 res = self.c.create_ephemeral_hidden_service(
                     {80: port}, await_publication=await_publication,
                     basic_auth=basic_auth, key_type=key_type,
-                    key_content=key_content)
+                    key_content=key_content, timeout=10)
```
Comment (arades79) [inline hyperdome_server/onion.py:581]: 10 seconds may be too short for users with a high latency connection or poor node locality.

**Review by arades79 — APPROVED** [review]
All changes are good. Fixes some bugs and works as well as can be expected for this stage.

Comments are suggestions for future PRs to fix.
app.py can also be deleted.

File: hyperdome_client/hyperdome_client.py (hunk 4 above)
Comment (stevenpitts) [reply hyperdome_client/hyperdome_client.py:206]: Flattened? What does that mean?

File: hyperdome_server/onion.py (hunk 6 above)
Comment (stevenpitts) [reply hyperdome_server/onion.py:581]: Definitely agree. We can increase the value by a lot, but the issue is that onion connection fails very often, so we need to retry a lot; I want to figure out why that's happening.

File: hyperdome_client/hyperdome_client.py (hunk 4 above)
Comment (arades79) [reply hyperdome_client/hyperdome_client.py:206]: less nesting, especially considering that the if statements are relatively complex expressions that aren't exceptionally clear as to their purpose. Breaking this out using the guard statement pattern or something else with an equivalent effect.

File: hyperdome_server/onion.py (hunk 6 above)
Comment (arades79) [reply hyperdome_server/onion.py:581]: likely being addressed by #39

**Pull request merged**